### PR TITLE
fix: TokenSender should not revert if limit reached

### PR DESCRIPTION
**Pull request by ramenforbreakfast**
So since we already had most of these testcases written for revert cases in the old `TokenSender` under `_addAmount`, I have simply moved the majority of these to `exceedsAccountLimit` tests. Because `exceedsAccountLimit` is a read-only function, and we need to advance timestamps for corner cases, I had to fix the `mineBlocks` utility that we never use.

In addition to these testcases, I have added some amount adding testcases to `_addAmount` since we now just want to check that amounts are added regardless of exceeding the limit and we already have covered situations when the account is reset.

### CODE REVIEW

File: packages/prepo-shared-contracts/test/PeriodicAccountLimits.test.ts [hunk 1]
```diff
@@ -110,190 +110,192 @@ describe('=> PeriodicAccountLimits', () => {
...
+      expect(await periodicAccountLimits.exceedsAccountLimit(user.address, 1)).eq(true)
+      // Set the period to the time of the last tx
+      await periodicAccountLimits.connect(deployer).setAccountLimitResetPeriod(15)
 
-      expect(await periodicAccountLimits.getAmountThisPeriod(user.address)).eq(TEST_LIMIT.sub(1))
+      expect(await periodicAccountLimits.exceedsAccountLimit(user.address, 1)).eq(false)
     })
 
-    it('adds amount if total = limit and period exactly reached', async () => {
+    it('evaluates total based on new period if period lengthened', async () => {
```
Comment (XavierEkkel) [inline packages/prepo-shared-contracts/test/PeriodicAccountLimits.test.ts:231]: for this one, would be better to test that it was true before and still true after period lengthened

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: packages/prepo-hardhat/src/smart-contracts-utils.ts [hunk 2]
```diff
@@ -104,7 +104,10 @@ function recordDeployment(envVarName: string, contract: Contract): void {
   process.env[envVarName] = contract.address
 }
 
-async function mineBlocks(provider: providers.Web3Provider, blocks: number): Promise<void> {
+async function mineBlocks(
```
[the remark this answers is not in the document]
Comment (ramenforbreakfast) [reply packages/prepo-hardhat/src/smart-contracts-utils.ts:107]: This is because we want to advance the time without actually executing a transaction, typically we use `setNextTimestamp` to prime the time when the _next_ transaction will run.
